### PR TITLE
[Shipping] Removed unneeded composite resolver interface

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets |
| License         | MIT

Change proposed in [this comment](https://github.com/Sylius/Sylius/pull/5216#issuecomment-225164083).

### CODE REVIEW

File: src/Sylius/Bundle/ShippingBundle/spec/Form/Type/ShippingMethodChoiceTypeSpec.php [hunk 1]
```diff
@@ -13,24 +13,20 @@
...
-use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * @author Arnaud Langlade <arn0d.dev@gamil.com>
  */
 class ShippingMethodChoiceTypeSpec extends ObjectBehavior
 {
     function let(
-        CompositeMethodsResolverInterface $compositeMethodsResolver,
+        MethodsResolverInterface $compositeMethodsResolver,
```
Comment (CarwynNelson) [inline src/Sylius/Bundle/ShippingBundle/spec/Form/Type/ShippingMethodChoiceTypeSpec.php]: This variable is called `$compositeMethodsResolver` but the interface is now `MethodsResolverInterface`. It's just a small thing but this now makes it seem like a composite methods resolver is expected to be injected in, even though that won't necessarily be true.

Similar occurrences elsewhere.

File: src/Sylius/Component/Shipping/spec/Resolver/CompositeMethodsResolverSpec.php [hunk 2]
```diff
@@ -38,7 +37,7 @@ function it_is_initializable()
 
     function it_implements_composite_methods_resolver_interface()
```
Comment (CarwynNelson) [inline src/Sylius/Component/Shipping/spec/Resolver/CompositeMethodsResolverSpec.php]: This function name should really be changed to represent the change in interface away from a composite methods resolver to a methods resolver.

File: src/Sylius/Component/Shipping/Resolver/CompositeMethodsResolver.php [hunk 3]
```diff
@@ -45,4 +45,12 @@ public function getSupportedMethods(ShippingSubjectInterface $shippingSubject)
 
         return [];
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(ShippingSubjectInterface $subject)
+    {
+        return true;
```
Comment (pamil) [inline src/Sylius/Component/Shipping/Resolver/CompositeMethodsResolver.php]: It supports a shipping subject only if one of nested resolvers supports it.

Comment (pamil) [inline src/Sylius/Component/Shipping/Resolver/CompositeMethodsResolver.php]: Anyway, it seems like this method is not used anywhere except `CompositeMethodsResolver::getShippingMethods()` and always returns true. We can remove it and then that method will return first not-empty result from nested methods resolver.

Comment (CarwynNelson) [inline src/Sylius/Component/Shipping/Resolver/CompositeMethodsResolver.php]: `ShippingMethodsByZonesAndChannelResolver` seems to be the only class which implements this interface that does not return true. Could there be a possibility that a separate interface is required for that class and `supports()` can be removed from the `MethodResolverInterface`? Not sure if I'm over thinking this as I've only just started diving into Sylius (just looking at some PR's to get the feel of it)

https://github.com/Sylius/Sylius/blob/db5a8dcd3b23fbaea73deaf80ddc6487ea5b60b9/src/Sylius/Component/Core/Resolver/ShippingMethodsByZonesAndChannelResolver.php

**Comment by pjedrzejewski** [comment]
@CarwynNelson Thanks for the review! :)

File: src/Sylius/Bundle/ShippingBundle/Form/Type/ShippingMethodChoiceType.php [hunk 4]
```diff
@@ -52,12 +50,12 @@ class ShippingMethodChoiceType extends AbstractType
...
 
     /**
-     * @param CompositeMethodsResolverInterface $compositeShippingMethodsResolver
+     * @param MethodsResolverInterface $compositeShippingMethodsResolver
      * @param ServiceRegistryInterface $calculators
      * @param RepositoryInterface $repository
      */
     public function __construct(
-        CompositeMethodsResolverInterface $compositeShippingMethodsResolver,
+        MethodsResolverInterface $compositeShippingMethodsResolver,
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ShippingBundle/Form/Type/ShippingMethodChoiceType.php]: Then we should also rename the variable.

File: src/Sylius/Bundle/ShippingBundle/spec/Form/Type/ShippingMethodChoiceTypeSpec.php (hunk 1 above)
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ShippingBundle/spec/Form/Type/ShippingMethodChoiceTypeSpec.php]: Same as above.

File: src/Sylius/Component/Shipping/Resolver/CompositeMethodsResolver.php (hunk 3 above)
Comment (Zales0123) [inline src/Sylius/Component/Shipping/Resolver/CompositeMethodsResolver.php]: @pamil @CarwynNelson the idea was, to unified all shipping methods resolvers with one ``MethodsResolverInterface``, to give possibility to inject both specific method resolver and composite resolver, which contains resolvers registry. Due to that fact, we must keep ``supports()`` method in all of them, as they can be used separately. I agree, that returning ``true`` here is nasty, but I can't came up with better option.

Comment (pamil) [inline src/Sylius/Component/Shipping/Resolver/CompositeMethodsResolver.php]:
```php
foreach ($this->resolversRegistry->all() as $resolver) {
     if ($resolver->supports($shippingSubject)) {
         return true;
     }
}

return false;
```

**Comment by pjedrzejewski** [comment]
Thank you Mateusz. :)

**Pull request merged**